vnext/Microsoft.ReactNative/Views/DevMenu.cpp

```diff
@@ -0,0 +1,286 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "pch.h"
+#include "DevMenu.h"
+#include "Modules/DevSettingsModule.h"
+#include "UI.Xaml.Controls.Primitives.h"
+#include "UI.Xaml.Controls.h"
+#include "UI.Xaml.Input.h"
+#include "UI.Xaml.Interop.h"
+#include "UI.Xaml.Markup.h"
+#include "Utils/Helpers.h"
+#include "IReactDispatcher.h"
+#include "winrt/Windows.UI.Core.h"
+#include "Views/KeyboardEventHandler.h"
+
+namespace Microsoft::ReactNative {
+
+React::ReactPropertyId<React::ReactNonAbiValue<std::shared_ptr<DevMenuManager>>> DevMenuManagerProperty() noexcept
+{
+  static React::ReactPropertyId<React::ReactNonAbiValue<std::shared_ptr<DevMenuManager>>> propId {
+      L"ReactNative.DevMenuManager", L"DevMenuManager" };
+  return propId;
+}
+
+/*static*/ void DevMenuManager::InitDevMenu(Mso::CntPtr<Mso::React::IReactContext> const &reactContext) noexcept
+{
+  auto devMenu = std::make_shared<DevMenuManager>(reactContext);
+  devMenu->Init();
+  React::ReactPropertyBag(reactContext->Properties()).Set(DevMenuManagerProperty(), devMenu);
+}
+
+void DevMenuManager::Init() noexcept {
```

Comment:
What's the motivation for two phase init? Seems like we're not delaying anything or reporting failure, so it would be nice if we could fold into the ctor and eliminate a small source of error.

Reply:
The code was originally a nativemodule (there is a devmenu module that we dont have implemented).  But I talked with FB and that module is going away.  This two phase init was left over from when it was a native module, which wouldn't have the reactcontext in the ctor.

Reply:
Actually weak_from_this doesn't work in the ctor.